src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -161,6 +161,14 @@ public boolean canProvision(Label label) {
                         node -> {
                           logger.info("Agent {} is online", name);
                           return node;
+                        })
+                    .exceptionally(
+                        ex -> {
+                          logger.warn(
+                              "Agent {} failed to come online with error {}", name, ex.getCause());
+                          mesosApi.killAgent(name);
+                          throw new RuntimeException(
+                              "Agent failed to come online within timeout period");
```

Comment:
I wonder if this is required since the future already failed. We need tests for this. Also, please at the cause to the `RuntimeException`. Otherwise this becomes untracable.